BHoM_Engine/Query/Geometry.cs

```diff
@@ -34,7 +34,7 @@ public static partial class Query
         /**** Public Methods                            ****/
         /***************************************************/
 
-        public static IGeometry IGeometry(this IBHoMObject obj)
+        public static IGeometry IGeometry(this IObject obj)
```

Comment:
I don't fully understand why but the change from `IBHoMObject` to `IObject` appears to have unexpected consequences when in the Rhino and GH UI. 
BHoM geometry no longer renders and a casting issue is introduced.
Recent experiments with changes to representation methods made the same change (to `IToRhino` method in Rhino_Toolkit) with the same consequences. 
Note the GH preview mode set to show all and the absence of coral coloured geometry in the model space in the image below.
![Screenshot 2021-02-18 081013](https://user-images.githubusercontent.com/6618854/108325508-0adfd480-71c1-11eb-9a58-a540cace5545.jpg)

Reply:
I unfortunately expected there could be things happening like this. I still had to test this properly, but due to how things are configured in the framework I think that https://github.com/BHoM/BHoM_Engine/issues/2316 is currently quite difficult to implement. I will have a good look, but will probably remove it from this PR.

Reply:
I have removed https://github.com/BHoM/BHoM_Engine/issues/2316 with https://github.com/BHoM/BHoM_Engine/pull/2308/commits/e3d1fb6ae3c462623d9054a2b9a3d1a0e2462903 and everything works now.